jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/handler/ConditionalHandler.java

```diff
@@ -0,0 +1,735 @@
+     * Exclude {@link InetAddressPattern} in the conditions to be met
+     * @param patterns String representations of {@link InetAddressPattern}s that are
+     *              tested against the {@link ConnectionMetaData#getRemoteSocketAddress() getRemoteSocketAddress()} of
+     *              {@link Request#getConnectionMetaData()}.
+     */
+    public void excludeInetAddressPattern(String... patterns)
+    {
+        for (String p : patterns)
+            exclude(InetAddressPattern.from(p));
+    }
+
+    /**
+     * {@link IncludeExclude#include(Object) Include} arbitrary {@link Predicate}s in the conditions.
+     * @param predicates {@link Predicate}s that are tested against the {@link Request}.
+     * This method is optimized so that a passed {@link MethodPredicate} or {@link PathSpecPredicate} is
+     * converted to a more efficient {@link #includeMethod(String...)} or {@link #include(PathSpec...)} respectively.
+     */
+    @SafeVarargs
+    public final void include(Predicate<Request>... predicates)
+    {
+        if (isStarted())
+            throw new IllegalStateException(getState());
+        for (Predicate<Request> p : predicates)
+        {
+            if (p instanceof MethodPredicate methodPredicate)
+                includeMethod(methodPredicate._method);
+            else if (p instanceof PathSpecPredicate pathSpecPredicate)
+                include(pathSpecPredicate._pathSpec);
+            else
+                _predicates.include(p);
+        }
+    }
+
+    /**
+     * {@link IncludeExclude#exclude(Object) Exclude} arbitrary {@link Predicate}s in the conditions.
+     * @param predicates {@link Predicate}s that are tested against the {@link Request}.
+     * This method is optimized so that a passed {@link MethodPredicate} or {@link PathSpecPredicate} is
+     * converted to a more efficient {@link #excludeMethod(String...)} or {@link #exclude(PathSpec...)} respectively.
+     */
+    @SafeVarargs
+    public final void exclude(Predicate<Request>... predicates)
+    {
+        if (isStarted())
+            throw new IllegalStateException(getState());
+        for (Predicate<Request> p : predicates)
+        {
+            if (p instanceof MethodPredicate methodPredicate)
+                excludeMethod(methodPredicate._method);
+            else if (p instanceof PathSpecPredicate pathSpecPredicate)
+                exclude(pathSpecPredicate._pathSpec);
+            else
+                _predicates.exclude(p);
+        }
+    }
+
+    private boolean testMethods(Request request)
+    {
+        return _methods.test(request.getMethod());
+    }
+
+    private boolean testPathSpecs(Request request)
+    {
+        return _pathSpecs.test(Request.getPathInContext(request));
+    }
+
+    private boolean testPredicates(Request request)
+    {
+        return _predicates.test(request);
+    }
+
+    @Override
+    protected void doStart() throws Exception
+    {
+        _handlePredicate = TypeUtil.truePredicate();
```

Comment:
This seems overkill, costs an allocation every time, and I doubt there is measured proof that is any faster, and it's more code to maintain.
Why not keeping it simple and leaving it at `null`?

Reply:
It is a small allocation once on doStart that makes the following code  simpler.  Without this, each of the last three setups become: 
```java
        if (!_predicates.isEmpty())
            _handlePredicate = _handlePredicate == null ? this::testPredicates : _handlePredicate.and(this::testPredicates);
``` 
which whilst not that complex is not as obvious at a casual glance.    For the tiny cost of allocating an instance with no data fields, I think the code is much clearer with the utility.